### PR TITLE
[flutter_tools] disable source maps by default for release builds, enable for run and with command line arg

### LINKED ISSUE #67328: Web builds should pass --no-source-maps to dart2js when built for production
While we do not output the source map file, the source map embedding is still in the file

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Disables source map production by default for `build web`. For web builds performed as a part of `flutter run --release`, enable the source maps, or allow force enabling with `--source-maps` command line flag.

fixes https://github.com/flutter/flutter/issues/67328

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build_web.dart [hunk 1]
```diff
@@ -37,6 +37,13 @@ class BuildWebCommand extends BuildSubCommand {
       help: 'Disable dynamic generation of code in the generated output. '
         'This is necessary to satisfy CSP restrictions (see http://www.w3.org/TR/CSP/).'
     );
+    argParser.addFlag(
+      'source-maps',
+      defaultsTo: false,
+      help: 'Whether to generate a sourcemap file. These can be used by browsers '
+      'To view the original source code of a compiled and minified Dart application. '
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/build_web.dart:44]: nit: "view and debug"?

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 2]
```diff
@@ -207,6 +211,8 @@ class Dart2JSTarget extends Target {
         '-Ddart.vm.product=true',
       for (final String dartDefine in decodeDartDefines(environment.defines, kDartDefines))
         '-D$dartDefine',
+      if (!sourceMapsEnabled)
+        '--no-source-maps',
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:215]: Should we pass `--source-maps` when `sourceMapsEnabled` for explicitness?

**Review by yjbanov — APPROVED** [review]
lgtm

File: packages/flutter_tools/lib/src/build_system/targets/web.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart:215]: --source-maps is not a flag, just --no-source-maps

File: packages/flutter_tools/lib/src/commands/build_web.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/build_web.dart]: Done

**Comment by jonahwilliams** [comment]
The tree is actually green, landing

**Pull request merged**